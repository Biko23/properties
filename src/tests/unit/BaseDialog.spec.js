import 'jsdom-global'
import { createLocalVue, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Dialog from '../../components/BaseDialog.vue'

describe('BaseDialog.vue', () => {
    const localVue = createLocalVue()
    let vuetify

    beforeEach(() => {
        vuetify = new Vuetify()
    })

    const mountFunction = options => {
        return mount(Dialog, {
          localVue,
          vuetify,
          ...options,
        })
    }

  it('should have a custom message and match snapshot', () => {
        const wrapper = mountFunction({
          propsData: { message: 'Test message' }
        })
    
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should have a custom title and match snapshot', () => {
        const wrapper = mountFunction({
            propsData: { title: 'Success' }
          })
        
        expect(wrapper.html()).toMatchSnapshot()
    })
})

