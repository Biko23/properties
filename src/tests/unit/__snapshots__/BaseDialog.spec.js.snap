// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDialog.vue should have a custom message and match snapshot 1`] = `
"<div role=\\"dialog\\" class=\\"v-dialog__container\\">
  <!---->
</div>"
`;

exports[`BaseDialog.vue should have a custom title and match snapshot 1`] = `
"<div role=\\"dialog\\" class=\\"v-dialog__container\\">
  <!---->
</div>"
`;
